core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java

```diff
@@ -191,6 +191,9 @@ private void declareIdempotentValidator(ExtensionDeclarer extensionDeclarer, Cla
         .withExpressionSupport(NOT_SUPPORTED)
         .describedAs("Defines the prefix of the object store names. This will only be used for the internally built object store.");
 
+    validator.withOptionalComponent("privateObjectStore")
```

Comment:
this is wrong. This is a parameter not a component. It is missing type, expression support, dsl model, and element reference. Look at `org.mule.runtime.extension.internal.loader.enricher.OAuthDeclarationEnricher.EnricherDelegate#addOAuthStoreConfigParameter` for inspiration